ex_schemas/schema_v3.json

```diff
@@ -0,0 +1,496 @@
+                         "pattern":"duffy_grp"
+                     },
+                     "res_defs":{
+                         "type": "array",
+                         "items" : {
+                             "oneOf" :
+                                 [
+                                   {"$ref": "#/definitions/duffy"}
+                                 ]
+                            }
+                    },
+                      "assoc_creds":{
+                        "description":"contains creds file associated to this resource",
+                        "type":"string"
+                    }
+                 },
+                 "required":["resource_group_name","res_group_type","res_defs","assoc_creds"],
+                 "additionalProperties": true
+         },
+         "libvirt_grp": {
+                 "type" : "object",
+                 "properties" : {
+                     "resource_group_name": {
+                         "type":"string"
+                     },
+                     "res_group_type": {
+                         "type":"string",
+                         "pattern":"libvirt_grp"
+                     },
+                     "res_defs":{
+                         "type": "array",
+                         "items" : {
+                             "oneOf" :
+                                 [
+                                   {"$ref": "#/definitions/libvirt"}
+                                 ]
+                            }
+                    },
+                      "assoc_creds":{
+                        "description":"contains creds file associated to this resource",
+                        "type":"string"
+                    }
+                 },
+                 "required":["resource_group_name","res_group_type","res_defs","assoc_creds"],
+                 "additionalProperties": true
+         },
+         "rackspace": {
+                 "type" : "object",
+                 "properties" : {
+                     "resource_group_name": {
+                         "type":"string"
+                     },
+                     "res_group_type": {
+                         "type":"string",
+                         "pattern":"rackspace"
+                     },
+                     "res_defs":{
+                         "type": "array",
+                         "items" : {
+                             "oneOf" :
+                                 [
+                                   {"$ref": "#/definitions/rax_server"}
+                                 ]
+                            }
+                    },
+                      "assoc_creds":{
+                        "description":"contains creds file associated to this resource",
+                        "type":"string"
+                    }
+                 },
+                 "required":["resource_group_name","res_group_type","res_defs","assoc_creds"],
+                 "additionalProperties": true
+         }
+    }
+}
+
```

Comment:
schema_v3 provides a strict division of resource group functionality , 
One bug which i found in schema_v2 was , when we declare a resource_group_type  = "openstack", it actually allows aws resources also to be defined inside that openstack group which kind of gives us a flexibility, but i felt inherently by having a resource_group_type its wrong. 
so if want to have a combination of multiple cloud resources , we can have a a new resource group type known as "hybrid" which accepts all the resource types.